samza-sql/src/main/java/org/apache/samza/sql/translator/JoinTranslator.java

```diff
@@ -250,7 +251,17 @@ private String dumpRelPlanForNode(RelNode relNode) {
         SqlExplainLevel.EXPPLAN_ATTRIBUTES);
   }
 
-  private SqlIOConfig resolveSourceConfig(RelNode relNode) {
+  private SqlIOConfig resolveSourceConfigForTable(RelNode relNode) {
+    if (relNode instanceof LogicalProject) {
+      return resolveSourceConfigForTable(((LogicalProject) relNode).getInput());
+    }
+
+    // For stream-table-table join, we will run into a join relNode. The output of stream-table join is a stream.
```

Comment:
This comment is not very clear.  Can you reword it a little bit. We are returning the sourceConfig for the table as null when the table is in the join rather than an output table, That's  because?

Reply:
Done!